modyn/tests/storage/internal/grpc/storage_service_impl_test.cpp

```diff
@@ -130,6 +131,7 @@ std::tuple<int64_t, int64_t> add_non_existing_sample(const YAML::Node& config, c
   if (!session.get_last_insert_id("samples", inserted_sample_id_ll)) {
     FAIL("Failed to insert sample into database");
   }
+  session.close();
```

Comment:
One additional comment here: The tests ran through without leaks beforehand, so I hope the destructor of the session would work correctly... but I hope that this might fix the leak issues you encountered.

Reply:
Yes it fixed and I think maybe the leak is only marked as error when the amount is over a threshold?

But the `session` **is** related to the memory leak. see https://github.com/eth-easl/modyn/blob/61d39e58dc0e24a93f1498c987e4689ed788efac/modyn/storage/include/internal/grpc/storage_service_impl.hpp#L123